File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int mul = 1;
        int zeros = 0;
        for(int x : nums){
            if(x!=0) mul *= x;
            else if(x==0) zeros++;
        }
        int ans[] = new int[nums.length];
        for(int i = 0 ;i< nums.length ;i++){
            if(zeros>1) ans[i]= 0;
            else if(zeros==1){
                if(nums[i]==0) ans[i] = mul ;
                else ans[i] =0;
            }
            else{
                ans[i] = mul/nums[i];
            }
        }
        return ans ;

    }
}  

File: Data Structures & Algorithms/combination-target-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @returns {number[][]}
     */
    ans = [];
    combinationSum(nums, target) {
         this.ans = []
         this.solve(nums,0,target, [])
         return this.ans
    }
    solve = function (nums , idx , target , curr){
          if(target==0){
            this.ans.push([...curr])
            return
          }
          if(target < 0 || nums.length == idx) return

          curr.push(nums[idx])
          this.solve(nums,idx,target-nums[idx],curr)
          curr.pop()
          this.solve(nums,idx+1,target,curr)
    }
}
